modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -416,12 +416,6 @@ public boolean onSnapshotLoad(Path path) {
 
     @Override
     public void onShutdown() {
-        // TODO: IGNITE-17958 - probably, we should not close the storage here as PartitionListener did not create the storage.
-        try {
-            storage.close();
```

Comment:
So the storage will be shut down either when the `TableStorage` is shutdown (on node stop), or when the partition is evicted from the node (in which case it will be not just shut down, but also destroyed)?

It would be great to have a test making sure that a non-evicted partition storage is closed on node stop.

Reply:
Storages will be closed when the `TableStorage` stops.
And are destroyed, on evict of partition or destruction of the table.

Found tests:
- org.apache.ignite.internal.table.distributed.TableManagerTest#endTableManagerStopTest
- org.apache.ignite.internal.table.distributed.TableManagerTest#testDropTable
- I don’t even know if testing is needed to evict a partition, it will be possible to add it during implementation [IGNITE-18160](https://issues.apache.org/jira/browse/IGNITE-18160,)